### ISSUE TITLE
AWS auto lightcurve uploads not happening

### ISSUE THREAD

**Issue by rachel3834** [pyDANDIA/pyDANDIA#144]
The pipeline is not uploading lightcurves to AWS from the automatic pipeline, though they are reaching MOP.

**Comment by rachel3834** [comment]
The $red_dir/logs/reduction_control log indicates that the pipeline is attempting the AWS upload when a lightcurve has been produced, and no error is reported.  Nevertheless, no lightcurve arrives at the AWS repo.  

This doesn't seem to be caused by a missing credentials or missing config issue.  I've tested the relevant code - aws_utils.py  - by running it from the commandline with the operational pipeline's auto_pipeline_config.json file and the lightcurve uploaded without problem and no errors were reported.  This suggests the issue is only happening in crontab operation. 

 I've improved the logging in this section of the code to see if I can see code output when the upload happens at the end of the automatic pipeline reduction as triggered under the crontab.

**Comment by rachel3834** [comment]
Added robonet_tool/data_harvest/upload_lcs_aws.py to bulk upload all lightcurves on local disk to AWS in the interim.

**Comment by rachel3834** [comment]
Updated logging for the automatic pipeline indicates that uploads are happening normally, to the right AWS bucket.  I've confirmed that the up to date lightcurve files are arriving.  I'm really not sure why this was causing an issue in the first place and it seems to be working now.  The only thing I did note is that the upload code deletes previous lightcurve versions for a given target before uploading the new one.  If that process somehow ran amok, conceivably the pipeline was removing but not uploading lightcurves.  But this is restricted to a single event, so it doesn't really explain the symptoms here.  

Closing this issue until we see a reoccurrence.

**Comment by rachel3834** [comment]
Believe I've figured out what is happening.  Valerio's modified version of my download software had an extra argument to selectively download only files matching the search term.  Adding this functionality lead me to realise that it wasn't finding files I knew to be in the bucket.  This is because the list_objects_v2 function returns a *paginated* list of entries, and we have crossed the pagination threshold in the number of lightcurve files.  

Problem solved by adding handing for pagination in the download software.
